pkg/asset/asset.go

```diff
@@ -26,6 +26,9 @@ type WritableAsset interface {
 
 	// Files returns the files to write.
 	Files() []*File
+
+	// SetState sets the asset data from an unmarshalled File slice
+	SetState([]*File) error
```

Comment:
What is the reason to add another function to the interface. Can you document it.

Reply:
Ack. Need to document the reason properly. Just to jot down here though:
We need a function that can restore the content of the WritableAsset when it has been unmarshalled from the state file. Since we unmarshal using a dynamic interface, this is a form of custom unmarshal based on object type.

Reply:
We can do the delayed unmarshal. That was one of the ideas that crossed my mind.
Another one was not to unmarshal the asset object itself. Just store the map[type.String][]*File just like that. And any asset can look up for its 'files' by indexing on its own type.String().

However, we would still need the constructor to be public and some method to load up the object's contents from outside (as I expect this would typically be used in the storeImpl.fetch() method).

Reply:
> We can do the delayed unmarshal.

It will allow us to no do this SetState extra functions. I would rather do this than add SetState on each asset.